sklearn/utils/extmath.py

```diff
@@ -174,6 +174,24 @@ def density(w, **kwargs):
     return d
 
 
+def _csr_to_array_fallback(X):
+    """ This provides the equivalent functionality to
+       csr_matrix.toarray()
+    and provides a fix for scipy versions <1.0.0 where
+    the above command fails for arrays with more than 2**31
+    elements.
+    """
+    if X.format != 'csr':
+        raise ValueError('Only CSR sparse arrays are supported')
```

Comment:
This is a private method with an explicit name and the only location where we use it in the code does the check on `X.format` explicitly. Let's get rid of this untested input check.

If we would like to keep the input check, it should be more informative (put the actual value of `X.format` in the error message) and should be tested.